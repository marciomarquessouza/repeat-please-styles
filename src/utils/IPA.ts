export const IPA = {
	ʌ: {
		symbol: 'ʌ',
		type: 'vowel',
		examples: ['cUp', 'lUck'],
	},
	'ɑ:': {
		symbol: 'ɑ:',
		type: 'vowel',
		examples: ['Arm', 'fAther'],
	},
	æ: {
		symbol: 'æ',
		type: 'vowel',
		examples: ['cAt', 'blAck'],
	},
	e: {
		symbol: 'e',
		type: 'vowel',
		examples: ['mEt', 'bEd'],
	},
	ə: {
		symbol: 'ə',
		type: 'vowel',
		examples: ['Away', 'cinemA'],
	},
	'ɜ:': {
		symbol: 'ɜ:',
		type: 'vowel',
		examples: ['tURn', 'lEARn'],
	},
	ɪ: {
		symbol: 'ɪ',
		type: 'vowel',
		examples: ['hIt', 'sITTING'],
	},
	'i:': {
		symbol: 'i:',
		type: 'vowel',
		examples: ['SEE', 'HEAT'],
	},
	ɒ: {
		symbol: 'ɒ',
		type: 'vowel',
		examples: ['HOT', 'ROCK'],
	},
	'ɔ:': {
		symbol: 'ɔ:',
		type: 'vowel',
		examples: ['CALL', 'FOUR'],
	},
	ʊ: {
		symbol: 'ʊ',
		type: 'vowel',
		examples: ['PUT', 'COULD'],
	},
	'u:': {
		symbol: 'u:',
		type: 'vowel',
		examples: ['BLUE', 'FOOD'],
	},
	aɪ: {
		symbol: 'aɪ',
		type: 'vowel',
		examples: ['FIVE', 'EYE'],
	},
	aʊ: {
		symbol: 'aʊ',
		type: 'vowel',
		examples: ['NOW', 'OUT'],
	},
	eɪ: {
		symbol: 'eɪ',
		type: 'vowel',
		examples: ['SAY', 'EIGHT'],
	},
	oʊ: {
		symbol: 'oʊ',
		type: 'vowel',
		examples: ['GO', 'HOME'],
	},
	ɔɪ: {
		symbol: 'ɔɪ',
		type: 'vowel',
		examples: ['BOY', 'JOIN'],
	},
	eə: {
		symbol: 'eə',
		type: 'vowel',
		examples: ['WHERE', 'AIR'],
	},
	ɪə: {
		symbol: 'ɪə',
		type: 'vowel',
		examples: ['NEAR', 'HERE'],
	},
	ʊə: {
		symbol: 'ʊə',
		type: 'vowel',
		examples: ['PURE', 'TOURIST'],
	},
};
