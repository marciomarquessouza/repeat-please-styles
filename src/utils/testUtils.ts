import { ShallowWrapper } from 'enzyme';

export const findByDataTest = (
	wrapper: ShallowWrapper,
	dataTest: string,
): ShallowWrapper => {
	return wrapper.find(`[data-test="${dataTest}"]`);
};
