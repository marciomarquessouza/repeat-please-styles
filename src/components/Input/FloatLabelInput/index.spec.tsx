import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { findByDataTest } from '../../../utils/testUtils';
import { FloatLabelInput } from './';

const setup = (props = {}): ShallowWrapper => {
	return shallow(<FloatLabelInput {...props} />);
};

describe('Float Label Input Basic Tests', () => {
	let wrapper: ShallowWrapper;

	beforeEach(() => {
		wrapper = setup();
	});

	it('should render the component properly', () => {
		const floatLabelInput = findByDataTest(wrapper, 'floatLabelInput');
		expect(floatLabelInput).toHaveLength(1);
	});

	it('should match to snapshot test', () => {
		const floatLabelInput = findByDataTest(wrapper, 'floatLabelInput');
		expect(floatLabelInput).toMatchSnapshot();
	});
});
