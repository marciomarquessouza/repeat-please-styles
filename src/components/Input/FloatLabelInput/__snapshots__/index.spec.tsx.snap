// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Float Label Input Basic Tests should match to snapshot test 1`] = `ShallowWrapper {}`;
